### PR TITLE
Form

### PR DESCRIPTION
**Pull request by laurellx**
uploading form

### CODE REVIEW

**Review by laurellx — COMMENTED** [review]
Reviewed

File: angular.json [hunk 1]
```diff
@@ -100,5 +100,8 @@
         }
       }
     }
+  },
+  "cli": {
+    "analytics": "2ea64b2c-de21-4550-8d36-f9063ce818bb"
```
Comment (miriamfraga) [inline angular.json:105]: angular.json

File: src/app/app.module.ts [hunk 2]
```diff
@@ -8,23 +12,27 @@ import { AppComponent } from './app.component';
 import { HeaderComponent } from './core/header/header.component';
 import { MainComponent } from './core/main/main.component';
 import { FooterComponent } from './core/footer/footer.component';
-import { SharedModule } from './pages/shared/shared.module';
+
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
```
Comment (miriamfraga) [inline src/app/app.module.ts:16]: app.module

File: src/app/app.module.ts [hunk 3]
```diff
@@ -8,23 +12,27 @@ import { AppComponent } from './app.component';
...
+
 
 @NgModule({
   declarations: [
     AppComponent,
     HeaderComponent,
     MainComponent,
-    FooterComponent
+    FooterComponent,
+    ManageComponent
```
Comment (miriamfraga) [inline src/app/app.module.ts:25]: delete manageComponent

File: src/app/app.module.ts [hunk 4]
```diff
@@ -8,23 +12,27 @@ import { AppComponent } from './app.component';
...
   ],
   imports: [
     BrowserModule,
     AppRoutingModule, 
     HttpClientModule,
     AppRoutingModule,
-
-    
-    
+    SharedModule,
```
Comment (miriamfraga) [inline src/app/app.module.ts:32]: imports of app module

File: src/app/models/models.ts [hunk 5]
```diff
@@ -4,5 +4,5 @@ export interface Item {
     price: number,
     description: string,
     star: number,
-    image:string,
+    image: string,
```
Comment (miriamfraga) [inline src/app/models/models.ts:7]: models interface

File: src/app/models/productRegistration.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+  
+    name: string;
+    price: number;
+    category: string[];
+    care: string;
+    music: string[];
+    description: string;
+    image: string;
+    rating: number;
+  }
```
Comment (miriamfraga) [inline src/app/models/productRegistration.ts:11]: interface

File: src/app/pages/manage/manage.component.html [hunk 7]
```diff
@@ -1 +1,43 @@
-<p>manage works!</p>
+<h2>Add product:</h2>
+
+<form novalidate (ngSubmit)="onSubmit()" [formGroup]="productRegistrationForm">
+    <fieldset>
+        <label for="name">Product name:</label>
```
Comment (miriamfraga) [inline src/app/pages/manage/manage.component.html:5]: form html

File: src/app/pages/manage/manage.component.ts [hunk 8]
```diff
@@ -1,14 +1,59 @@
+import { ProductRegistration } from './../../models/productRegistration';
 import { ProductService } from './../../service/product.service';
 import { Component } from '@angular/core';
```
Comment (miriamfraga) [inline src/app/pages/manage/manage.component.ts:3]: validators. manage.ts

File: src/app/pages/manage/manage.component.ts [hunk 9]
```diff
@@ -1,14 +1,59 @@
...
   templateUrl: './manage.component.html',
   styleUrls: ['./manage.component.scss']
 })
+
+
 export class ManageComponent {
- productRegistrationForm!: FormGroup;
-   constructor( private productBuilder:FormBuilder,
-   private productService:ProductService) {}
+  public productRegistrationForm!: FormGroup;
```
Comment (miriamfraga) [inline src/app/pages/manage/manage.component.ts:14]: arrays manage.ts

File: src/app/pages/manage/manage.module.ts [hunk 10]
```diff
@@ -1,15 +1,17 @@
 import { NgModule } from '@angular/core';
 import { CommonModule } from '@angular/common';
 
-import { ManageComponent } from './manage.component';
+import { ManageRoutingModule } from './manage-routing.module';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
 
 
 @NgModule({
```
Comment (miriamfraga) [inline src/app/pages/manage/manage.module.ts:8]: manage.module

File: src/app/service/product.service.ts [hunk 11]
```diff
@@ -4,11 +4,24 @@ import { HttpClient } from '@angular/common/http';
 import { Item } from '../models/models';
 
 const backUrl = "https://my-json-server.typicode.com/franlindebl/shopeame-api-v2/products"
+
 @Injectable()
 export class ProductService {
```
Comment (miriamfraga) [inline src/app/service/product.service.ts:10]: structure service array/interface

**Review by miriamfraga — APPROVED** [review]

**Comment by miriamfraga** [comment]
changes approved

**Pull request closed without merging**